File: AtmLocator/app/src/main/java/piotrek/atmlocator/AtmLocatorApplication.java
package piotrek.atmlocator;

import android.app.Application;

import piotrek.atmlocator.di.ApplicationComponent;
import piotrek.atmlocator.di.ApplicationModule;
import piotrek.atmlocator.di.DaggerApplicationComponent;

/**
 * Created by Admin on 2016-04-26.
 */
public class AtmLocatorApplication extends Application {

    public static ApplicationComponent component;

    public AtmLocatorApplication(){

        component = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();

    }
}

File: AtmLocator/app/src/main/java/piotrek/atmlocator/AtmLocatorApplication.kt
package piotrek.atmlocator

import android.app.Application

import piotrek.atmlocator.di.ApplicationComponent
import piotrek.atmlocator.di.ApplicationModule
import piotrek.atmlocator.di.DaggerApplicationComponent

/**
 * Created by Admin on 2016-04-26.
 */
class AtmLocatorApplication : Application() {


    override fun onCreate() {
        super.onCreate()

        component = DaggerApplicationComponent.builder().applicationModule(ApplicationModule(this)).build()
    }

    companion object {

        @JvmStatic lateinit  var component: ApplicationComponent
    }
}
